objects.md

```diff
@@ -0,0 +1,118 @@
+# About
+
+This repo needs to represent the following information:
+
+1. Adding/removing versions to channels
+2. Remove or add edges to all channels
+3. Remove or add edges to a specific channel
+4. Implement an "upgrade policy" object specifically for y-streams
+
+Remember that cininnatti should pull most edges from the release manifest. This data should typiclly remove edges, not add them.
+
+# Directory Structure
+
+We do this via a combination of objects and directory structure. All object should be YAML (so we can make comments). The directories in the repo should be:
+- /nodes/
+- /channels/
+- /added_edges/
+- /blocked_edges/
+- /y-policies/
+
+# Nodes
+
+The /nodes objects are very simple. It is not strictly necessary as long as cincinnatti pulls all releases from quay. However it is useful for a CI tool to build a final (smaller) graph and for disconnected customers to build a complete cincinnatti graph. (though the later is not the first target goal!) So /nodes/4.1.24 would look like:
+```
+version: 4.1.24
+payload: quay.io/openshift-release-dev/ocp-release@sha256:6f87fb66dfa907db03981e69474ea3069deab66358c18d965f6331bd727ff23f
+```
+
+# Channels
+
+The /channels object described the nodes in a channel. For example the /channels/stable-4.1.yaml file would contain
+
+```
+name: stable-4.1
+versions:
+  - 4.1.0
+  - 4.1.1
+  - ...
+  - 4.1.11
+  # 4.1.12 Was excluded from stable!!!
+  - 4.1.13
+  - 4.1.14
+subchannels:
+```
+
+Subchannels can be implemented later. The idea is that inside stable-4.2.yaml we would have:
+```
+subchannels:
+  - stable-4.1
+```
+Which would pull all nodes from stable-4.1 into stable-4.2. Without having to explicitly add them.
+
+# Added Edges
+
+Adding edges will be rare. It will have the same format as removed edges. It will typiclly be used only in conjunction with 'policy' below.
+
+# Blocked Edges
+
+Blocked edges will be used any time we find a problem. The blocked edge object for /blocked_edges/4.1.10.yaml would looke like:
+```
+to: 4.1.10
+- channels:
+    - *
+  from: *
```

Comment:
Definitely need this, e.g. to exclude 4.1.z->4.2.2 without excluding 4.2.0->4.2.2.  But I'd rather use a regexp or explicit list instead of this glob (or maybe it's a single special character, and not a glob?).

Reply:
I was thinking of `*` as a single special character, not a glob.  If we need anything other than a single character I'm fine with a regex, not glob.